### ISSUE TITLE
Sorting problem after upgrade

### ISSUE THREAD

**Issue by gbouteiller** [rofrischmann/fela#598]
**Type**: Bug

| Packages | Version |
| -------- | ------- |
| fela     | 6.2.0   |

-------------------
#### Description
After upgrading to the last version of fela, I came across a weird issue : the css that I rendered statically were not ordered anymore.

#### Proposal
 After investigation, I found that it was due to the sort method in clusterCache in fela-utils that was returning 0 which leaded to unpredictable order. So, I added the key indexes to be compared instead of returning 0, forcing the original ordering in that case and it works.

BEFORE : 
```javascript
// fela-utils/src/sortCache.js
export default function sortCache(cache: Object, sort: Function): Object {
  const sortedKeys = Object.keys(cache).sort((left, right) =>
    sort(cache[left], cache[right])
  )
}
```
AFTER : 
```javascript
// fela-utils/src/sortCache.js
export default function sortCache(cache: Object, sort: Function): Object {
  const sortedKeys = arrayMap( arrayMap( Object.keys(cache), (key, index) =>({index, key})).sort((left, right) => {
    const sorting = sort(cache[left.key], cache[right.key]);
    return sorting !== 0 ? sorting : left.index - right.index;
  }), ({key}) => key);
}
```

**Comment by rofrischmann** [comment]
Hmm, can you give me some example code? I actually tested that behaviour and it seems fine to me. Returning 0 in a sort function won't reorder the array right?

**Comment by gbouteiller** [comment]
It happened in an existing application, so I'll try to reproduce it with a less complex example and give you a link. 
About returning 0, in fact, it can reorder the array and that is the problem. I got the info from [Mozilla documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) : 

> If compareFunction(a, b) returns 0, leave a and b unchanged with respect to each other, but sorted with respect to all different elements. **Note: the ECMAscript standard does not guarantee this behaviour, and thus not all browsers (e.g. Mozilla versions dating back to at least 2003) respect this.**

In fact, when I replaced your code by always returning 0 and checked the keys, there was kind of a dichotomy process as I got the element n/2 + 1 as my first element then my first element as second, ...

**Comment by gbouteiller** [comment]
After further investigation, I was wondering why it was working on my simple example and not in my nuxt application and in fact, it was... till the number of keys to sort was superior to 10. 
It turns out that the problem only appears on **server side** because of **nodeJS** implementation of  `sort` : [Here](https://www.linkedin.com/pulse/javascript-v8-sorting-algorithm-behind-scene-arraysort-mazahar-shaikh) is the explanation : 

> It have combination of InsertionSort and QuickSort.
It goes for InsertionSort if the array length is short (length <= 10) otherwise QuickSort takes you to the party :)

To sum up, using sort ( that internally uses QuickSort ) with more than 10 keys on server side and returning 0 can change the initial order of your keys and mess up the css importing order. Voila.

**Comment by rofrischmann** [comment]
Oh my... so shipping a custom sort solution should do the job as well right?

**Comment by gbouteiller** [comment]
I think it will do, yeah :)

**Comment by rofrischmann** [comment]
Alright, got a solution ready (even better one) that I’ll release today or tomorrow.
Am on my way to vacation right now = no internet in the car
